source/dagger/gameplay/platformer/platformer_controller.cpp

```diff
@@ -27,19 +27,50 @@ void PlatformerControllerSystem::SpinUp()
 void PlatformerControllerSystem::Run()
 {
     Engine::Registry().view<InputReceiver, Sprite, Animator, PlatformerCharacter>().each(
-        [](const InputReceiver input_, Sprite& sprite_, Animator& animator_, const PlatformerCharacter& char_)
+        [](const InputReceiver input_, Sprite& sprite_, Animator& animator_, /*const*/ PlatformerCharacter& char_)
         {
+            Float32 down=input_.values.at("down");
             Float32 run = input_.values.at("run");
-            if (run == 0)
+            Float32 useP = input_.values.at("use");
+            Float32 block = input_.values.at("block");
+            Float32 sAttack = input_.values.at("heavy");
+
+            if(down && run)        
+            {          
+                sprite_.position.x += char_.speed * sprite_.scale.x * Engine::DeltaTime();
+                AnimatorPlay(animator_, "souls_like_knight_character:ROLL");
+            } 
+            else if (sAttack != 0)
             {
-                AnimatorPlay(animator_, "souls_like_knight_character:IDLE");
+                AnimatorPlay(animator_, "souls_like_knight_character:ATTACK");
+                
+            }
+            else if (block != 0)
+            {
+                AnimatorPlay(animator_, "souls_like_knight_character:BLOCK");
+                
             }
-            else
+            else if (useP != 0)
             {
+                AnimatorPlay(animator_, "souls_like_knight_character:DRINK_POTION");
+            } 
+            else if (run != 0)
+            { 
                 AnimatorPlay(animator_, "souls_like_knight_character:RUN");
                 sprite_.scale.x = run;
                 sprite_.position.x += char_.speed * sprite_.scale.x * Engine::DeltaTime();
             }
+
+            if(run==0 && useP==0 && down==0 && block==0 && sAttack==0) 
+            {
+                AnimatorPlay(animator_, "souls_like_knight_character:IDLE");
+            }
+            
```

Comment:
Here's a thought: could I trick this system into enabling me to cancel a block/attack/jump half way and begin another animation?

Reply:
It could probably, but I must admit I'm not sure how exactly to implement it.